components/global/global.js

```diff
@@ -5,38 +5,49 @@ import {
   unmountComponentAtNode,
 } from 'react-dom';
 
+function setGlobalNoscroll(flag) {
+  const body = global.window.document.body;
+  flag
+    ? body.classList.add('ui-global_noscroll')
+    : body.classList.remove('ui-global_noscroll');
+}
+
 /**
  * Global component
  * React component for transportation of modals, lightboxes, loading bars... to document.body
  */
 class Global extends Component {
   componentDidMount() {
-    if (this.props.noscroll) {
-      global.window.document.body.classList.add('ui-global_noscroll');
-    }
-
     this.target = global.window.document.createElement('div');
     this.target.classList.add('ui-global');
     global.window.document.body.appendChild(this.target);
-    this.componentDidUpdate();
+    this.updateContainerAndBody(false);
   }
 
-  componentDidUpdate() {
-    unstable_renderSubtreeIntoContainer(this, (
-      <div className={this.props.className}>
-        {this.props.children}
-      </div>
-    ), this.target);
+  componentDidUpdate(prev) {
+    this.updateContainerAndBody(prev.noscroll);
   }
 
   componentWillUnmount() {
     if (this.props.noscroll) {
-      global.window.document.body.classList.remove('ui-global_noscroll');
+      setGlobalNoscroll(false);
```

Comment:
Say that you have 2 components on the page:

`Modal` and `SlidingPanel`, both using `Global`, both with `noscroll=true`.

If `Modal` _unmounts_ and `SlidingPanel` is still _mounted_, can you tell me which classes will the `body` DOM Node have?